include/circt/Support/LoweringOptions.h

```diff
@@ -90,6 +90,9 @@ struct LoweringOptions {
   /// columns.
   enum { DEFAULT_LINE_LENGTH = 90 };
   unsigned emittedLineLength = DEFAULT_LINE_LENGTH;
+
+  /// Add an explicit bitcast for avoiding bitwidth mismatch LINT errors.
+  bool explicitBitcastAddMul = false;
```

Comment:
don't add expression state to the emitter object.  if you need to, thread it through the visitor just like binaryflags and precedence are.